homeassistant/components/control4/__init__.py

```diff
@@ -0,0 +1,275 @@
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_TOKEN,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import device_registry as dr, entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_CONFIG_LISTENER,
+    CONF_CONTROLLER_NAME,
+    CONF_DIRECTOR,
+    CONF_DIRECTOR_ALL_ITEMS,
+    CONF_DIRECTOR_MODEL,
+    CONF_DIRECTOR_SW_VERSION,
+    CONF_DIRECTOR_TOKEN_EXPIRATION,
+    CONF_LIGHT_COLD_START_TRANSITION_TIME,
+    CONF_LIGHT_TRANSITION_TIME,
+    DEFAULT_LIGHT_COLD_START_TRANSITION_TIME,
+    DEFAULT_LIGHT_TRANSITION_TIME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["light"]
+
+
+async def async_setup(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Stub to allow setting up this component.
+
+    Configuration through YAML is not supported at this time.
+    """
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Control4 from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN].setdefault(entry.entry_id, {})
+
+    config = entry.data
+    account = C4Account(config[CONF_USERNAME], config[CONF_PASSWORD])
+    try:
+        await account.getAccountBearerToken()
+    except client_exceptions.ClientError as exception:
+        _LOGGER.error("Error connecting to Control4 account API: %s", exception)
+        raise PlatformNotReady()
+    hass.data[DOMAIN][entry.entry_id][CONF_ACCOUNT] = account
+
+    controller_name = config[CONF_CONTROLLER_NAME]
+    hass.data[DOMAIN][entry.entry_id][CONF_CONTROLLER_NAME] = controller_name
+
+    director_token_dict = await account.getDirectorBearerToken(controller_name)
+    director = C4Director(config[CONF_HOST], director_token_dict[CONF_TOKEN])
+    hass.data[DOMAIN][entry.entry_id][CONF_DIRECTOR] = director
+    hass.data[DOMAIN][entry.entry_id][
+        CONF_DIRECTOR_TOKEN_EXPIRATION
+    ] = director_token_dict["token_expiration"]
+
+    # Add Control4 controller to device registry
+    controller_href = (await account.getAccountControllers())["href"]
+    hass.data[DOMAIN][entry.entry_id][
+        CONF_DIRECTOR_SW_VERSION
+    ] = await account.getControllerOSVersion(controller_href)
+
+    result = re.search("_(.*)_", controller_name)
+    hass.data[DOMAIN][entry.entry_id][CONF_DIRECTOR_MODEL] = result.group(1).upper()
+
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, controller_name)},
```

Comment:
The identifier needs to be stable and something the user cannot change such as a serial number of mac address.  It should conform to the unique id requirements.   

https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

Reply:
`controller_name` is assigned by the Control4 API based on MAC address - as far as I know, it cannot be changed by the user.